pkgs/desktops/gnome/extensions/material-shell/default.nix

```diff
@@ -1,25 +1,42 @@
-{ stdenv, lib, fetchFromGitHub, gnome }:
-
-stdenv.mkDerivation rec {
-  pname = "gnome-shell-extension-material-shell";
-  version = "12";
-
-  src = fetchFromGitHub {
+{ stdenv, pkgs, lib, fetchFromGitHub, nix-gitignore, gnome, glib, nodejs }:
+let
+  ver = "40.a";
+  source = fetchFromGitHub {
     owner = "material-shell";
     repo = "material-shell";
-    rev = version;
-    sha256 = "0ikrh70drwr0pqjcdz7l1ky8xllpnk7myprjd4s61nqkx9j2iz44";
+    rev = ver;
+    sha256 = "Plzjid2X6C5NM/OWaMk78FKUwNT1taubFBtYw1dNIRw=";
   };
+  nodeDependencies = (import ./node-composition.nix {
+    inherit pkgs;
+    inherit (stdenv.hostPlatform) system;
+  }).nodeDependencies.override (oa: {
+    src = source;
+  });
+in stdenv.mkDerivation rec {
+  pname = "gnome-shell-extension-material-shell";
+  version = ver;
 
-  # This package has a Makefile, but it's used for building a zip for
-  # publication to extensions.gnome.org. Disable the build phase so
-  # installing doesn't build an unnecessary release.
-  dontBuild = true;
+  src = source;
+
+  buildInputs = [
+    glib.dev
+    nodejs
+    nodeDependencies
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+    ln -s "${nodeDependencies}/lib/node_modules" ./node_modules
+    export PATH="${nodeDependencies}/bin:$PATH"
+    make compile
```

Comment:
`make compile` uses `npx` which gets build tools that don't exist in `node_modules`.

In this case, `npx` tries to download `node-sass`, `tsc`, `rollup` and the build results in a failure.
Since `node-sass` is already deprecated and doesn't exist in the latest nixpkgs, we need to provide `node-sass` using `nodePackages.sass`:

```nix
{
  nativeBuildInputs = [
    glib.dev
    nodejs
    nodeDependencies
    # add
    nodePackages.rollup
    nodePackages.typescript
    (pkgs.writeShellScriptBin "node-sass" ''
      ${nodePackages.sass}/bin/sass "$@"
    '')
  ];
}
```